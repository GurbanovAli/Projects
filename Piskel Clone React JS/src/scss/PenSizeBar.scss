@import './Presets.scss';

.pen-size {
  display: flex;
  flex-shrink: 0;
  flex-direction: column;
  align-items: center;
  width: 60px;
  margin-right: -20px;
  color: azure;
  img {

  }
  .size-one {
    width: 10px;
    height: 10px;
    padding-top: 30%;
  }
  .size-two {
    width: 15px;
    height: 15px;
    padding-top: 25%;
  }
  .size-three {
    width: 20px;
    height: 20px;
    padding-top: 16%;
  }
  .size-four {
    width: 25px;
    height: 25px;
    padding-top: 7%;
  }
  .square {
    width: 30px;
    height: 30px;
    margin-top: 5px;
    border: 1px solid wheat;
    text-align: center;
    &:first-child {
      margin-top: 0;
    }
  }
  .activeTool {
    border: 3px solid $specialColor;
  }

}
