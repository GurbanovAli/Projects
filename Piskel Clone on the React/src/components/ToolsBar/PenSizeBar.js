import React, { Component } from 'react';
import size from '../../Assets/icons/square.svg';

export default class PenSizeBar extends Component {
    render () {
        const penSizes = [
            { size: 1, className: 'size-one' },
            { size: 2, className: 'size-two' },
            { size: 3, className: 'size-three' },
            { size: 4, className: 'size-four' }
        ];

        return (
            <div className="pen-size">
                {penSizes.map(pen => (
                    <div
                        key={pen.className}
                        className={`square ${this.props.penSize === pen.size ? 'activeTool' : ''}`}
                        onClick={() => this.props.changePenSize(pen.size)}>
                        <img src={size} className={pen.className} />
                    </div>
                ))}
            </div>
        );
    }
}
