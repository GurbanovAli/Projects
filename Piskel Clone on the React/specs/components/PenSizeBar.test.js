import React from 'react';
import { create } from 'react-test-renderer'

import PenSizeBar from '../../src/components/ToolsBar/PenSizeBar';

describe('PenSizeBar', () => {
  it('snapshot', () => {
    const tree = create(<PenSizeBar />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
