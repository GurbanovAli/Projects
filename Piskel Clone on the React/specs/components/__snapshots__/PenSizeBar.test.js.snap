// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PenSizeBar snapshot 1`] = `
<div
  className="pen-size"
>
  <div
    className="square activeTool"
  >
    <img
      className="size-one"
      src="test-file-stub"
    />
  </div>
  <div
    className="square activeTool"
  >
    <img
      className="size-two"
      src="test-file-stub"
    />
  </div>
  <div
    className="square activeTool"
  >
    <img
      className="size-three"
      src="test-file-stub"
    />
  </div>
  <div
    className="square activeTool"
  >
    <img
      className="size-four"
      src="test-file-stub"
    />
  </div>
</div>
`;
